src/sentry/monitors/tasks.py

```diff
@@ -46,7 +52,21 @@ def _get_monitor_checkin_producer() -> KafkaProducer:
     return KafkaProducer(build_kafka_configuration(default_config=producer_config))
 
 
+def _get_monitor_checkin_partitions() -> Mapping[int, PartitionMetadata]:
+    topic = settings.KAFKA_INGEST_MONITORS
+    cluster_name = get_topic_definition(topic)["cluster"]
+
+    conf = get_kafka_admin_cluster_options(cluster_name)
+    admin_client = AdminClient(conf)
+    result = admin_client.list_topics(topic=topic)
+    topic_metadata = result.topics.get(topic)
+
+    assert topic_metadata
+    return topic_metadata.partitions
+
+
 _checkin_producer = SingletonProducer(_get_monitor_checkin_producer)
+_checkin_partitions = _get_monitor_checkin_partitions()
```

Comment:
Hmm are you sure you want this to run on import of this module? Shouldn't it happen on the first time we need to use it then be cached?

Reply:
Yeah good question. Probably don't want this happening all the time.

Is there a simple util for cache on first call? I think there's a `@cached` decorator or something right?

Reply:
Using `lru_cache`